Add flag to skip (second) final image build phase

purpleidea opened the pull request:
This adds a new flag to cause the (second) final stage to be skipped.
This is useful for when you're more interested in the build artifacts
being built in the correct environment, and less interested in the final
image.

purpleidea commented:
In combination with https://github.com/systemd/mkosi/pull/363 this makes `mkosi` very useful for combining with fpm to build packages in the correct OS environments.

lucasdemarchi commented on mkosi:
```diff
@@ -4422,7 +4424,10 @@ def build_stuff(args: CommandLineArguments) -> None:
+    if args.skip_final_phase:
+        print_step('Skipping (second) final image build phase.')
+    else:
+        raw, tar, root_hash = build_image(args, workspace, do_run_build_script=False, cleanup=True)
```
But you're not skipping the steps below... In your usecase you may also want to make sure type is set to directory.

purpleidea replied:
I'm not sure I understand your comment, but I tested the patch and it seems to work as I expected. Did I miss something?

keszybz replied:
`raw`, `tar`, `root_squash` are used unconditionally right below. If this call to `build_image` is skipped, they might not be set at all.

purpleidea commented:
Hi, can we merge this? Thanks!

poettering commented:
also, needs rebase

purpleidea commented:
@keszybz 

> raw, tar, root_squash are used unconditionally right below. If this call to build_image is skipped, they might not be set at all.

Good catch, thank you. Fixed.

@poettering 

> also, needs rebase

Rebased to current git master.

Thank you!

purpleidea commented:
\o/ Thanks!
